components/movementsensor/gpsutils/vrs.go

```diff
@@ -66,3 +85,79 @@ func HasVRSStream(sourceTable *Sourcetable, mountPoint string) (bool, error) {
 
 	return stream.Nmea, nil
 }
+
+// Close closes the VRS connection and any other background threads.
+func (vrs *VRS) Close() error {
+	vrs.workers.Stop()
```

Comment:
If you construct a `VRS` but don't call `StartGGAThread`, then `vrs.workers` will be nil and this line will panic.

If you construct a `VRS` and call `Close` in one thread while calling `StartGGAThread` in another at the same time, this line might stop the old workers and then new workers are started, and we'll leak goroutines.

Both of these problems could be avoided if `StartGGAThread` were made a private function that is only called once from within `ConnectToVirtualBase()`. Would that make semantic sense? In other words, could `ConnectToVirtualBase()` take a `getGGA` argument? Could we start the thread before doing all the other stuff in `getNtripFromVRS()`?

Reply:
so two things
1. I think we could have ConnectToVIrtualBase() take a getGGA. Having the GGA is a requirement for using VRS, so it seems reasonable to add that to the constructor
2. I think I would rather move the "other stuff" in getNtripFromVRS() into ConnectToVirtualBase(). I don't want to send gga messages before the connection is actually established, and ensuring the connection is actually established seems important enough to include in the constructor. 

TY for the callout here!